static/extensions/MubiLop/penguingpt.js

```diff
@@ -0,0 +1,752 @@
+        }
+
+        singlePrompt(args) {
+            const prompt = args.PROMPT;
+
+            return Scratch.fetch(api_url, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        'Origin': 'https://gptcall.net/',
+                        'Referer': 'https://gptcall.net/'
+                    },
+                    body: JSON.stringify({
+                        model: this.model,
+                        messages: [{
+                            role: "user",
+                            content: prompt
+                        }]
+                    }),
+                })
+                .then(response => {
+                    if (!response.ok) {
+                        throw new Error(`Network response was not ok: ${response.status} ${response.statusText}`);
+                    }
+                    return response.json();
+                })
+                .then(data => {
+                    const botResponse = data.choices[0].message.content;
+                    return botResponse;
+                })
+                .catch(error => {
+                    console.error("Error sending prompt to GPT", error.message);
+                    return "Error: ", error.message;
+                });
+        }
+
+        generateImage(args) {
+            const prompt = args.PROMPT;
+            const requestedModel = args.MODEL
+
+            return Scratch.fetch("https://reverse.mubi.tech/image/generate", {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify({
+                        model: requestedModel,
+                        prompt: prompt
+                    }),
+                })
+                .then(response => {
+                    if (!response.ok) {
+                        throw new Error(`Network response was not ok: ${response.status} ${response.statusText}`);
+                    }
+                    return response.json();
+                })
+                .then(data => {
+                    let botResponse
+                    if (requestedModel === "dalle-3") {
+                        botResponse = JSON.stringify(data);
+                    } else {
+                        botResponse = data.results
+                    }
+                    return botResponse;
+                })
+                .catch(error => {
+                    console.error("Error sending prompt to Image Generator", error.message);
+                    return "Error: ", error.message;
+                });
+        }
+        generateImageAndImport(args) {
+            const prompt = args.PROMPT;
+            const requestedModel = args.MODEL
+            const Name = args.NAME || `AIGenerated_${prompt}`;
+
+            return Scratch.fetch("https://reverse.mubi.tech/image/generate", {
```

Comment:
Does not respect `api_url` variable, either disable the ability to change it or use the variable here